test/__init__.py

```diff
@@ -39,6 +39,24 @@
 LONG_TIMEOUT = 0.5 if os.environ.get("CI") else 0.01
 
 
+def _can_resolve(host):
+    """ Returns True if the system can resolve host to an address. """
+    can_resolve = False
+
+    try:
+        socket.getaddrinfo(host, None, socket.AF_UNSPEC)
+        can_resolve = True
+    except socket.gaierror:
+        pass
+
+    return can_resolve
```

Comment:
Wouldn't the following be simpler?

```suggestion
    try:
        socket.getaddrinfo(host, None, socket.AF_UNSPEC)
        return True
    except socket.gaierror:
        return False
```